objcgen/objcgenerator.cs

```diff
@@ -796,8 +796,124 @@ protected override void Generate (ProcessedMethod method)
 				GenerateDefaultValuesWrappers (objcsig, method.Method);
 		}
 
+		void ReturnArrayValue (Type t)
+		{
+			var typecode = Type.GetTypeCode (t);
+			implementation.WriteLine ("MonoArray* __resarr = (MonoArray *) __result;");
+			implementation.WriteLine ("int __resarrlength = mono_array_length (__resarr);");
+
+			if (typecode == TypeCode.Byte)
+				implementation.WriteLine ("NSMutableData* __resobj = [NSMutableData dataWithCapacity:__resarrlength];");
+			else {
+				implementation.WriteLine ("id __autoreleasing* __resarrbuf = (id __autoreleasing *) malloc (sizeof (id) * __resarrlength);");
```

Comment:
any reason for `__autoreleasing` ?
ARC wants something for I would have used `__strong` since you're doing the allocation (it's not a variable that some other code will allocate) and it's usage is all local